components/automate-ui/src/app/pages/api-token/details/api-token-details.component.html

```diff
@@ -29,20 +29,30 @@
         </chef-tab-selector>
       </chef-page-header>
     <section class="page-body">
-      <form [formGroup]="updateNameForm">
+      <form [formGroup]="updateForm">
         <chef-form-field>
           <label>Name *</label>
-          <input chefInput formControlName="name" type="text" (keyup)="handleNameChange()">
+          <input chefInput formControlName="name" type="text">
+          <chef-error *ngIf="updateForm.get('name').hasError('required') && updateForm.get('name').dirty">
+            Name is required.<!-- drop period? -->
+          </chef-error>
+          <!-- error for NON_BLANK validator? -->
+        </chef-form-field>
+        <chef-form-field>
+          <label>Status *</label>
+          <chef-phat-radio ngDefaultControl formControlName="status">
+            <chef-option value="active">Active</chef-option>
+            <chef-option value="inactive">Inactive</chef-option>
+          </chef-phat-radio>
         </chef-form-field>
       </form>
-      <chef-button primary [disabled]="!updateNameForm.valid || disableSave || saveInProgress" (click)="saveNameChange()">
+      <div id="save">
+      <chef-button primary [disabled]="!updateForm.valid || !updateForm.dirty || saveInProgress" (click)="saveChange()">
         <chef-loading-spinner *ngIf="saveInProgress"></chef-loading-spinner>
-        <ng-container *ngIf="!saveInProgress">
-          <span *ngIf="disableSave">Saved</span>
-          <span *ngIf="!disableSave">Save</span>
-        </ng-container>
-        <span *ngIf="saveInProgress">Saving...</span>
+        Save
       </chef-button>
+      <span id="saved-note" *ngIf="!firstLoad && !updateForm.dirty">All changes saved.</span>
```

Comment:
Try this:

1. edit
2. Save (All changes saved.)
3. Add a char ()
4. Delete that char (All changes saved.)

Is that a good thing? Not sure. The alternative is the way it is done in team-details and project-details, as a result of a subscription (look for `saveSuccessful`). But your approach here is more straightforward IMO.

Reply:
This was introduced with your last commit, which did the `markAsPristine()`, I think.

When I left this, the formgroup would never go back to _pristine_ unless it was reset -- when the save was successful.

Raised this in slack, let's ensure we've discussed the desired behaviour here.